src/topology/metric_space/thickened_indicator.lean

```diff
@@ -0,0 +1,211 @@
+@[simps] def thickened_indicator {δ : ℝ} (δ_pos : 0 < δ) (E : set α) : α →ᵇ ℝ≥0 :=
+{ to_fun := λ (x : α), (thickened_indicator_aux δ E x).to_nnreal,
+  continuous_to_fun := begin
+    apply continuous_on.comp_continuous
+            continuous_on_to_nnreal (continuous_thickened_indicator_aux δ_pos E),
+    intro x,
+    exact (lt_of_le_of_lt (@thickened_indicator_aux_le_one _ _ δ E x) one_lt_top).ne,
+  end,
+  map_bounded' := begin
+    use 2,
+    intros x y,
+    rw [nnreal.dist_eq],
+    apply (abs_sub _ _).trans,
+    rw [nnreal.abs_eq, nnreal.abs_eq, ←one_add_one_eq_two],
+    have key := @thickened_indicator_aux_le_one _ _ δ E,
+    apply add_le_add;
+    { norm_cast,
+      refine (to_nnreal_le_to_nnreal ((lt_of_le_of_lt (key _) one_lt_top).ne) one_ne_top).mpr
+             (key _), },
+  end, }
+
+lemma thickened_indicator.coe_fn_eq_comp {δ : ℝ} (δ_pos : 0 < δ) (E : set α) :
+  ⇑(thickened_indicator δ_pos E) = ennreal.to_nnreal ∘ thickened_indicator_aux δ E := rfl
+
+lemma thickened_indicator_le_one {δ : ℝ} (δ_pos : 0 < δ) (E : set α) (x : α) :
+  thickened_indicator δ_pos E x ≤ 1 :=
+begin
+  rw [thickened_indicator.coe_fn_eq_comp],
+  simpa using (to_nnreal_le_to_nnreal thickened_indicator_aux_lt_top.ne one_ne_top).mpr
+    (thickened_indicator_aux_le_one δ E x),
+end
+
+lemma thickened_indicator_one_of_mem_closure
+  {δ : ℝ} (δ_pos : 0 < δ) (E : set α) {x : α} (x_mem : x ∈ closure E) :
+  thickened_indicator δ_pos E x = 1 :=
+by rw [thickened_indicator_apply,
+       thickened_indicator_aux_one_of_mem_closure δ E x_mem, one_to_nnreal]
+
+lemma thickened_indicator_one {δ : ℝ} (δ_pos : 0 < δ) (E : set α) {x : α} (x_in_E : x ∈ E) :
+  thickened_indicator δ_pos E x = 1 :=
+thickened_indicator_one_of_mem_closure _ _ (subset_closure x_in_E)
+
+lemma thickened_indicator_zero
+  {δ : ℝ} (δ_pos : 0 < δ) (E : set α) {x : α} (x_out : x ∉ thickening δ E) :
+  thickened_indicator δ_pos E x = 0 :=
+by rw [thickened_indicator_apply, thickened_indicator_aux_zero δ_pos E x_out, zero_to_nnreal]
+
+lemma thickened_indicator_mono {δ₁ δ₂ : ℝ}
+  (δ₁_pos : 0 < δ₁) (δ₂_pos : 0 < δ₂) (hle : δ₁ ≤ δ₂) (E : set α) :
+  ⇑(thickened_indicator δ₁_pos E) ≤ thickened_indicator δ₂_pos E :=
+begin
+  intro x,
+  apply (to_nnreal_le_to_nnreal thickened_indicator_aux_lt_top.ne
+         thickened_indicator_aux_lt_top.ne).mpr,
+  apply thickened_indicator_aux_mono hle,
+end
+
+lemma thickened_indicator_subset {δ : ℝ} (δ_pos : 0 < δ) {E₁ E₂ : set α} (subset : E₁ ⊆ E₂) :
+  ⇑(thickened_indicator δ_pos E₁) ≤ thickened_indicator δ_pos E₂ :=
+λ x, (to_nnreal_le_to_nnreal thickened_indicator_aux_lt_top.ne
+      thickened_indicator_aux_lt_top.ne).mpr (thickened_indicator_aux_subset δ subset x)
+
+lemma thickened_indicator_tendsto_indicator_closure
+  {δseq : ℕ → ℝ} (δseq_pos : ∀ n, 0 < δseq n) (δseq_lim : tendsto δseq at_top (𝓝 0)) (E : set α) :
+  tendsto (λ n, (thickened_indicator (δseq_pos n) E).to_fun) at_top
```

Comment:
please use a coercion instead of `to_fun`.

Reply:
I will. I'm still rather new to these issues, and I'd like to understand why we prefer the `coe` over the `to_fun`... Is there a simple explanation? That would be useful both when I write further code myself, and so that I can recommend better  choices to others. Thank you!

Reply:
The idea is that everything should have one standard form, so that we don't need to write several versions of the same lemma. Typically, the standard form is the one that simp will converge to, and use. When we have a coercion, we tend to use it as the standard form, as opposed to implementation details such as `to_fun`, that should essentially never appear in a lemma statement.